### ISSUE TITLE
Directly include Base64-encoded images in .qmd

### ISSUE THREAD

**Issue by cscheid** [quarto-dev/quarto-cli#6568]
Pandoc is appending .png at the end of data URIs. I thought we had fixed this but apparently it's back.

### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/6567

<div type='discussions-op-text'>

<sup>Originally posted by **zeileis** August 21, 2023</sup>
### Description

When I include Base64-encoded images directly in .qmd files, they are not rendered correctly. When I just do

```
![](data:image/png;base64,i [...snip...] =)
```

Then a .png suffix is added at the end in the `<img =src"..">` which does not work.

Additionally, in some contexts I get the `file name too long (os error 36)` error, similar to https://github.com/quarto-dev/quarto-cli/issues/4613.

A minimal example replicating the first issue using quarto 1.3.450 and R package [base64enc](https://CRAN.R-project.org/package=base64enc) is:

````qmd
Output of `image(volcano)` in Base64:
```{r}
#| echo: false
#| results: "asis"
## create PNG file
img <- tempfile(fileext = ".png")
png(img)
image(volcano)
dev.off()

## write as base 64
img64 <- sprintf("![](%s)", base64enc::dataURI(file = img, mime = "image/png"))
writeLines(img64)
```
````

I wasn't able to pinpoint exactly when I get the `file name too long` error. I only get it with Opengraph/Twitter enabled in `_quarto.yml` but only in my actual website and not in a minimal fresh website. I can try to dig deeper in case this is needed.</div>

**Comment by cscheid** [comment]
This is coming from the Pandoc reader, so we can't undo it in Lua.

```
$ cat pandoc-base64-test.md
![](data:image/png;base64,ithisisabadencoding)

![](data:image/png;base64,ithisisabadencoding.png)
$ quarto pandoc pandoc-base64-test.md --default-image-extension=png -t html
<p><img src="data:image/png;base64,ithisisabadencoding.png" /></p>
<p><img src="data:image/png;base64,ithisisabadencoding.png" /></p>
```


Possibilities:

- We can try to work around it in qmd-reader.lua but I would prefer to not touch that file unless absolutely necessary; it's extremely fragile
- we can try to undo the damage in a Lua filter, but it won't be exact: we might remove intentional `.png` extensions. I don't know why someone would add a `.png` extension to a base64-encoded data URI (since that's invalid in base64 encodings) but it makes me uncomfortable to add one more non-invertible transformation to our pipeline.

We cou

**Comment by zeileis** [comment]
Thanks for confirming the problem and boiling it down to the `--default-image-extension=png` option. I had tried `pandoc` before my report, but not with that option. Should we report this upstream to the `pandoc` project as well?

**Comment by cscheid** [comment]
@zeileis I've already asked @tarleb who is a Pandoc developer. I'll update here when I know more.

**Comment by cderv** [comment]
> Should we report this upstream to the pandoc project as well?

Did we report this upstream already ? It seems wrong behavior from pandoc 🤔

**Comment by cscheid** [comment]
I haven't - go ahead if you want to. I'd rather fix now and remove the need to fix it later.

**Comment by cderv** [comment]
I'll do it. 
And 100% with you on fixing it right now. Just want to be sure we pass through to their team some behavior that we found.

**Comment by zeileis** [comment]
That's great, thanks!

**Comment by cderv** [comment]
FWIW @cscheid this has been solve in current dev version of Pandoc : https://github.com/jgm/pandoc/commit/d62e5f33dbc9472f138a4f886f0d16019336326f

Should be available next time we update our Pandoc version bundled.

**Comment by cderv** [comment]
This is available in Pandoc 3.1.9 (https://github.com/jgm/pandoc/releases/tag/3.1.9) and will be available in Quarto when this version will be included. Follow up on changelog to know when.
